src/cargo/core/compiler/context/mod.rs

```diff
@@ -565,10 +565,19 @@ impl Links {
             let pkg = unit.pkg.package_id();
 
             let describe_path = |pkgid: PackageId| -> String {
-                let dep_path = resolve.path_to_top(&pkgid);
-                let mut dep_path_desc = format!("package `{}`", dep_path[0]);
-                for dep in dep_path.iter().skip(1) {
-                    write!(dep_path_desc, "\n    ... which is depended on by `{}`", dep).unwrap();
+                let dep_path = resolve.graph().path_to_top(&pkgid);
+                let mut dep_path_desc = format!("package `{}`", pkgid);
+                for &(dep, req) in dep_path.iter() {
+                    let req = req.first().unwrap();
+                    write!(
+                        dep_path_desc,
+                        "\n    ... selected to fulfill the requirement \
```

Comment:
```suggestion
                        "\n    ... selected to satisfy the requirement \
```

"Satisfying" dependencies is already a term that a lot of people are familiar with,
even (or especially) when English isn't their first language.
Using this would probably make it more clear.

(just randomly found this PR, sorry for sort of necromancing, in case this has been resolved elsewhere)